modules/machine-lifecycle-hook-deletion.adoc

```diff
@@ -0,0 +1,79 @@
++
+In the context of machine deletion, the machine controller performs the following actions:
+--
+* Drain the node that is backed by the machine.
+* Delete the machine instance from the cloud provider.
+* Delete the `Node` object.
+--
+
+Lifecycle hook:: A defined point in the reconciliation lifecycle of an object where the normal lifecycle process can be interrupted. Components can use a lifecycle hook to inject changes into the process to accomplish a desired outcome.
++
+There are two lifecycle hooks in the machine `Deleting` phase:
+--
+* `preDrain` lifecycle hooks must be resolved before the node that is backed by the machine can be drained.
+* `preTerminate` lifecycle hooks must be resolved before the instance can be removed from the infrastructure provider.
+--
+
+Hook-implementing controller:: A controller, other than the machine controller, that can interact with a lifecycle hook. A hook-implementing controller can do one or more of the following actions:
++
+--
+* Add a lifecycle hook.
+* Respond to a lifecycle hook.
+* Remove a lifecycle hook.
+--
++
+Each lifecycle hook has a single hook-implementing controller, but a hook-implementing controller can manage one or more hooks.
+
+[id="machine-lifecycle-hook-deletion-order_{context}"]
+== Machine deletion processing order
+
+In {product-title} {product-version}, there are two lifecycle hooks for the machine deletion phase: `preDrain` and `preTerminate`. When all hooks for a given lifecycle point are removed, reconciliation continues as normal.
+
+.Machine deletion flow
+image::310_OpenShift_machine_deletion_hooks_0223.png["The sequence of events in the machine `Deleting` phase."]
+
+The machine `Deleting` phase proceeds in the following order:
+
+. An existing machine is slated for deletion for one of the following reasons:
+** A user with `cluster-admin` permissions uses the `oc delete machine` command.
+** The machine gets a `machine.openshift.io/delete-machine` annotation.
+** The machine set that manages the machine marks it for deletion to reduce the replica count as part of reconciliation.
+** The cluster autoscaler identifies a node that is unnecessary to meet the deployment needs of the cluster.
+** A machine health check is configured to replace an unhealthy machine.
+. The machine enters the `Deleting` phase, in which it is marked for deletion but is still present in the API.
+. If a `preDrain` lifecycle hook exists, the hook-implementing controller that manages it does a specified action.
++
+Until all `preDrain` lifecycle hooks are satisfied, the machine status condition `Drainable` is set to `False`.
+. There are no unresolved `preDrain` lifecycle hooks and the machine status condition `Drainable` is set to `True`.
+. The machine controller attempts to drain the node that is backed by the machine.
```

Comment:
Same wonder re: colon or not, leading into sub-bullets. ... This is probably fine/correct.